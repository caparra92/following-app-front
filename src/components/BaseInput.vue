<template>
  <ion-row>
    <ion-col>
      <ion-item class="container">
        <input
          v-bind="$attrs"
          :placeholder="label"
          :value="modelValue"
          class="field"
          @input="$emit('update:modelValue', $event.target.value)"
        />
      </ion-item>
    </ion-col>
  </ion-row>
</template>
<script setup>
import { IonRow, IonCol, IonItem, IonIcon } from '@ionic/vue';

const props = defineProps({
  label: {
    type: String,
    default: "",
  },
  modelValue: {
    type: [String, Number],
    default: "",
  }
});
</script>
<style scoped>
.container {
  width: 100%;
  padding: 0;
  margin: 0;
}

 .field {
    width: 100%;
    text-transform: lowercase;
    background-color: var(--ion-color-secondary);
    color: var(--ion-color-primary);
    height: 2.3em;
    border: 1px solid rgba(var(--ion-color-primary-rgb), 1);
    border-radius: 7px;
    outline: none;
    margin: .8em;
    transition: all 0.30s ease-in-out;
  }

  .field:focus {
    box-shadow: 0 0 3px rgba(var(--ion-color-primary-rgb), 1);
    border: 2px solid rgba(var(--ion-color-primary-rgb), 1);
  }
</style>
