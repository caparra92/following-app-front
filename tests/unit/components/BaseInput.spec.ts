import { mount, VueWrapper } from "@vue/test-utils";
import BaseInput from "@/components/BaseInput.vue";


describe('Base Input component', () => {
    let wrapper: VueWrapper;
    beforeEach(() => {
        wrapper = mount(BaseInput, {
            props: {
                label: 'Email',
                modelValue: 'caparra@mail.com'
            }   
        });
    });

    test('Should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should verify that component exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    test('Should define props in the component', () => {
        expect(wrapper.props().label).toBe('Email');
        expect(wrapper.props().modelValue).toBe('caparra@mail.com');
    });

    test('Should verify emmited event', () => {
        const input = wrapper.find('input');
        input.trigger('input');
        expect(wrapper.emitted('update:modelValue')?.length).toBe(1);
    });
});