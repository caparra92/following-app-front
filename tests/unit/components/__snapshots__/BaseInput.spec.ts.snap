// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Input component Should match with snapshot 1`] = `
<ion-row class="">
  <ion-col class="">
    <ion-item class=""><input placeholder="Email" class="field"></ion-item>
  </ion-col>
</ion-row>
`;
